clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1707,15 +1723,14 @@ public DeleteTopicsResult deleteTopics(final Collection<String> topicNames,
             final long now = time.milliseconds();
             final long deadline = calcDeadlineMs(now, options.timeoutMs());
             final Call call = getDeleteTopicsCall(options, topicFutures, validTopicNames,
-                Collections.emptyMap(), now, deadline);
+                    Collections.emptyMap(), now, deadline);
             runnable.call(call, now);
         }
-        return new DeleteTopicsResult(new HashMap<>(topicFutures));
+        return new HashMap<>(topicFutures);
```

Comment:
Seems we could use `Map<String, KafkaFuture<Void>>` as the type of `topicFutures`?

Reply:
We can't create a KafkaFuture object as it is abstract.

Reply:
I don't think this works because we use this map in getDeleteTopicsCall where we can't use completeExceptionally on a KafkaFuture object.

Reply:
Fair enough